daemon/graphdriver/driver.go

```diff
@@ -85,6 +86,26 @@ type Driver interface {
 	DiffSize(id, parent string) (size int64, err error)
 }
 
+// CustomImageAccessor is an interface to allow callers to query the object for
+// any implementation specific image information that might need to be inserted into
+// the graph.
+type CustomImageAccessor interface {
+	// GetCustomImages returns a json string with metadata describing the images
+	// known to this implementation. The data is a json encoded CustomImageInfoList.
+	GetCustomImages() (string, error)
```

Comment:
Specifically, I'd rather expose the Go type, which can be changed, rather than a JSON format which may be inflexible.